From 73f33a7578a31afb94382ad9d19c6cc4b44efcaa Mon Sep 17 00:00:00 2001
From: kfx <kfx@rock-chips.com>
Date: Thu, 1 Aug 2013 15:53:01 +0800
Subject: [PATCH] rk3026: disable adc keyboard

because rk280a adc reference voltage is 3.3V, but
rk30xx mainbord key's supply voltage is 2.5V and
rk31xx mainbord key's supply voltage is 1.8V.
---
 arch/arm/mach-rk3026/board-rk3026-tb.c  | 14 ++++++++++++++
 arch/arm/mach-rk3026/board-rk3028a-tb.c | 14 ++++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/arch/arm/mach-rk3026/board-rk3026-tb.c b/arch/arm/mach-rk3026/board-rk3026-tb.c
index a2ce0ec549d2..73910aaabf9e 100755
--- a/arch/arm/mach-rk3026/board-rk3026-tb.c
+++ b/arch/arm/mach-rk3026/board-rk3026-tb.c
@@ -434,6 +434,19 @@ static struct sensor_platform_data mma8452_info = {
 #include <plat/key.h>
 
 static struct rk29_keys_button key_button[] = {
+        {
+                .desc   = "play",
+                .code   = KEY_POWER,
+                .gpio   = PLAY_ON_PIN,
+                .active_low = PRESS_LEV_LOW,
+                .wakeup = 1,
+        },
+/* disable adc keyboard,
+ * because rk280a adc reference voltage is 3.3V, but
+ * rk30xx mainbord key's supply voltage is 2.5V and
+ * rk31xx mainbord key's supply voltage is 1.8V.
+ */
+#if 0 
 #ifdef RK31XX_MAINBOARD_V1
         {
                 .desc   = "vol-",
@@ -528,6 +541,7 @@ static struct rk29_keys_button key_button[] = {
 		.active_low = PRESS_LEV_LOW,
 	},
 #endif
+#endif
 };
 
 struct rk29_keys_platform_data rk29_keys_pdata = {
diff --git a/arch/arm/mach-rk3026/board-rk3028a-tb.c b/arch/arm/mach-rk3026/board-rk3028a-tb.c
index 390a35cd8734..2114a474f4db 100755
--- a/arch/arm/mach-rk3026/board-rk3028a-tb.c
+++ b/arch/arm/mach-rk3026/board-rk3028a-tb.c
@@ -434,6 +434,19 @@ static struct sensor_platform_data mma8452_info = {
 #include <plat/key.h>
 
 static struct rk29_keys_button key_button[] = {
+        {
+                .desc   = "play",
+                .code   = KEY_POWER,
+                .gpio   = PLAY_ON_PIN,
+                .active_low = PRESS_LEV_LOW,
+                .wakeup = 1,
+        },
+/* disable adc keyboard,
+ * because rk280a adc reference voltage is 3.3V, but
+ * rk30xx mainbord key's supply voltage is 2.5V and
+ * rk31xx mainbord key's supply voltage is 1.8V.
+ */
+#if 0 
 #ifdef RK31XX_MAINBOARD_V1
         {
                 .desc   = "vol-",
@@ -528,6 +541,7 @@ static struct rk29_keys_button key_button[] = {
 		.active_low = PRESS_LEV_LOW,
 	},
 #endif
+#endif
 };
 
 struct rk29_keys_platform_data rk29_keys_pdata = {
-- 
2.35.3

